externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala

```diff
@@ -50,7 +50,7 @@ abstract class SparkOperation(session: Session)
 
   private val timeZone: ZoneId = {
     spark.conf.getOption(TIMEZONE_KEY).map { timeZoneId =>
-      ZoneId.of(timeZoneId.replaceFirst("(\\+|\\-)(\\d):", "$10$2:"), ZoneId.SHORT_IDS)
+      ZoneId.of(timeZoneId.replaceFirst("([+\\-])(\\d):", "$10$2:"), ZoneId.SHORT_IDS)
```

Comment:
```scala
  def getZoneId(timeZoneId: String): ZoneId = {
    val formattedZoneId = timeZoneId
      // To support the (+|-)h:mm format because it was supported before Spark 3.0.
      .replaceFirst("(\\+|\\-)(\\d):", "$10$2:")
      // To support the (+|-)hh:m format because it was supported before Spark 3.0.
      .replaceFirst("(\\+|\\-)(\\d\\d):(\\d)$", "$1$2:0$3")

    ZoneId.of(formattedZoneId, ZoneId.SHORT_IDS)
  }
```

refer to upstream changes, we may change here with the above code

Reply:
Is there any doc for referring these upstream changes?

Reply:
no

Reply:
all right. But I could not understan why or in what circumstance that the replacement should be `"$10$2:"` rather than padding 0 in the front as  `"0$1$2:"`

Reply:
OK, LGTM after reading the comments again. Has adopted your code suggestion.